src/main/java/mate/academy/service/UserService.java

```diff
@@ -15,6 +17,11 @@ public class UserService {
      * Return <code>null</code> if there is no suitable user
      */
     public User findByEmail(String email) {
-        return null;
+        try {
+            return Arrays.stream(users).filter(user -> user.getEmail().equals(email))
+              .findFirst().get();
+        } catch (NoSuchElementException exception) {
+            return null;
+        }
```

Comment:
It's very cool that you are familiar with streams already, but this is not a proper way to handle `NoSuchElementException`, first of all, it is a Runtime exception, which means that we should not catch it even if we can, and we should avoid the situation that produces it, so following our method logic we should return null in case we did not find a user, so we can use `orElse()` method of `Optional` like that
```suggestion
        return Arrays.stream(users)
            .filter(user -> user.getEmail().equals(email))
            .findFirst()
            .orElse(null);
```
there will be a whole lot more information about streams, optionals, exceptions and other related things later in the course, so I would recommend to avoid using things that we haven't covered yet, unless you are sure in what you are doing.
Expected solution here would be just using a for each loop and comparing emails